src/common/seed/images.ts

```diff
@@ -81,18 +81,31 @@ const generateRandomImage = () => {
 	return selectedVariant;
 };
 
-const generateBuild = () => {
+const remainingStatuses = ['building', 'error', 'ready']; //Is it correct way to make global variable in that function?
+const generateBuild = (statusIndex?: number) => {
 	const statuses = ['building', 'error', 'ready'];
 	const randomNum = Math.round((Math.random() * 2));
 
+	if (statusIndex !== undefined) {
+		const randomStatusIndex = Math.round((Math.random() * (remainingStatuses.length - 1)));
+		const pickedStatus = remainingStatuses[randomStatusIndex];
+		remainingStatuses.splice(randomStatusIndex, 1);
```

Comment:
side effect. your functions must be simple and clean as that possible.

don't use a method with mutation of your initial data.